src/cpu/aarch64/acl_thread_proxy.hpp

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+* Copyright 2025 Arm Ltd. and affiliates
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_ACL_THREAD_PROXY_HPP
+#define CPU_AARCH64_ACL_THREAD_PROXY_HPP
+
+// The purpose of the proxy header file is exactly to enable system_header
```

Comment:
I am not sure about this pattern in general. As you can see from the build logs there are multiple ACL headers that have `Wundef` hits. Surely we would end up with a large number of `*_proxy.hpp` headers?

Reply:
I will let @jondea and @theComputeKid weigh in but neither proliferating a large number of `_proxy.hpp` files nor a single large `acl_proxy.hpp` seem ideal.

Reply:
@dzarukin [I have raised a patch](https://review.mlplatform.org/c/ml/ComputeLibrary/+/14388) on the ComputeLibrary side to address this and will hopefully have an update for you soon. If it gets merged perhaps we only enable this warning for non-AArch64 systems for now and then turn it back on when we bump ComputeLibrary. Let me know your thoughts.

Reply:
Let me explore if it's an easy thing to do. Dropping it for Aarch64 is an option for me.

Reply:
Looks like platform.cmake has necessary support so I dropped the flag for AARCH64 and proxy headers altogether.

Reply:
@dzarukin The fix has been merged in ACL. We should be able to enable when we next bump the version.